tests/test_replay.py

```diff
@@ -0,0 +1,129 @@
+        neuron_ids = np.concatenate(neuron_ids)
+
+        if random:
+            neuron_ids = np.random.permutation(neuron_ids)
+
+        return spikes, neuron_ids, sequence_epochs
+
+    def test_simulate_sequential_spikes(self):
+        """Test the spike simulation function."""
+        task_spikes, task_neurons, task_seq_epochs = self.simulate_sequential_spikes(
+            nneurons=self.nneurons,
+            minseqduration=self.minseqduration,
+            maxseqduration=self.maxseqduration,
+            duration=self.duration,
+            random=False,
+        )
+
+        # Check if the number of spikes is correct
+        self.assertEqual(len(task_spikes), self.nneurons * len(task_seq_epochs))
+
+        # Check if neuron IDs are within expected range
+        self.assertTrue(np.all(np.isin(task_neurons, np.arange(self.nneurons))))
+
+    def test_pairwise_bias_analysis(self):
+        """Test the PairwiseBias analysis."""
+        task_spikes, task_neurons, task_seq_epochs = self.simulate_sequential_spikes(
+            nneurons=self.nneurons,
+            minseqduration=self.minseqduration,
+            maxseqduration=self.maxseqduration,
+            duration=self.duration,
+            random=False,
+        )
+
+        post_spikes_sig, post_neurons_sig, post_sig_seq_epochs = (
+            self.simulate_sequential_spikes(
+                nneurons=self.nneurons,
+                minseqduration=self.minseqduration,
+                maxseqduration=self.maxseqduration,
+                duration=self.duration,
+                random=False,
+            )
+        )
+
+        post_spikes_nonsig, post_neurons_nonsig, post_nonsig_seq_epochs = (
+            self.simulate_sequential_spikes(
+                nneurons=self.nneurons,
+                minseqduration=self.minseqduration,
+                maxseqduration=self.maxseqduration,
+                duration=self.duration,
+                random=True,
+            )
+        )
+
+        transformer = PairwiseBias()
+
+        # Analyze significant replay
+        z_score_sig, p_value_sig, cosine_val_sig = transformer.fit_transform(
+            task_spikes,
+            task_neurons,
+            task_seq_epochs,
+            post_spikes_sig,
+            post_neurons_sig,
+            post_sig_seq_epochs,
+        )
+
+        # Analyze non-significant replay
+        z_score_nonsig, p_value_nonsig, cosine_val_nonsig = transformer.transform(
+            post_spikes_nonsig, post_neurons_nonsig, post_nonsig_seq_epochs
+        )
+
+        # Check results are of expected shape and values are valid
+        self.assertEqual(len(z_score_sig), len(post_sig_seq_epochs))
+        self.assertEqual(len(z_score_nonsig), len(post_nonsig_seq_epochs))
+
+        # Check that significant replay has lower p-values than non-significant replay
+        self.assertTrue(np.all(p_value_sig < 0.05))
+        self.assertTrue(np.mean(p_value_sig) < np.mean(p_value_nonsig))
```

Comment:
I'm still a bit confused why a random sequence would occasionally come up as more sequential than chance levels

Reply:
just accounting for stochasticity due to small number of shuffles. If we increase num_shuffles that discrepancy goes away but just to be on the safe side this remains a more robust test case